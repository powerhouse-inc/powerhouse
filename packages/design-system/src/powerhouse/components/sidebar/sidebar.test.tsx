import { render, screen } from "@testing-library/react";

import { Sidebar, SidebarFooter, SidebarHeader, SidebarPanel } from ".";

describe("Sidebar Component", () => {
  it("should match snapshot", () => {
    const { asFragment } = render(
      <Sidebar collapsed={false} maxWidth="300px" minWidth="100px">
        <SidebarPanel>
          <SidebarHeader />
        </SidebarPanel>
        <SidebarFooter />
      </Sidebar>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render correctly", () => {
    render(
      <Sidebar
        collapsed
        data-testid="sidebar"
        maxWidth="300px"
        minWidth="100px"
      />,
    );
    expect(screen.getByTestId("sidebar")).toBeInTheDocument();
  });

  it("should render expanded", () => {
    render(
      <Sidebar
        collapsed={false}
        data-testid="sidebar"
        maxWidth="300px"
        minWidth="100px"
      />,
    );
    expect(screen.getByTestId("sidebar")).toHaveStyle({ width: "300px" });
  });

  it("should render collapsed", () => {
    render(
      <Sidebar
        collapsed
        data-testid="sidebar"
        maxWidth="300px"
        minWidth="100px"
      />,
    );
    expect(screen.getByTestId("sidebar")).toHaveStyle({ width: "100px" });
  });
});
