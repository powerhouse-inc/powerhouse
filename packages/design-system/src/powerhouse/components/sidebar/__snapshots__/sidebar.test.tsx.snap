// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sidebar Component > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="group expanded flex h-full flex-col overflow-hidden transition-none duration-300 animate-in fade-in shadow-sidebar"
    style="width: 300px;"
  >
    <div
      class="no-scrollbar flex-1 overflow-auto text-gray-900 transition-shadow"
    >
      <div
        class="shrink-0"
      />
    </div>
    <div
      class="shrink-0"
    />
  </div>
</DocumentFragment>
`;
